quai/caip10.md

```diff
@@ -0,0 +1,224 @@
+
+```sh
+^quai:(9|15000):0x[a-fA-F0-9]{40}$
+```
+
+**Address Format**:
+
+```sh
+0x[PB][LB][34 hex chars]
+```
+
+Where:
+
+- `PB` = Prefix byte encoding shard location `(region << 4) | zone`
+- `LB` = Ledger byte (bit 0/first bit indicates ledger: 0=Quai, 1=Qi)
+- Remaining 18 bytes = Standard address data
+
+**Ledger Detection**:
+
+- Third hexadecimal character `0-7` → Quai ledger (account-based)
+- Third hexadecimal character `8-F` → Qi ledger (UTXO-based)
+
+**Prefix Byte Calculation**:
+
+```sh
+prefix_byte = (region << 4) | zone
+```
+
+**Valid Examples**:
+
+- `quai:9:0x00841c4349dbC3CD16098ef0aF2780BA0F7f5C02` (mainnet, cyprus-1 shard)
+- `quai:15000:0x004d2Ee5f59c0A7Df5C54d6C9d5a4f3e3b8E1d2F` (testnet, cyprus-1 shard)
+
+### Resolution Mechanics
+
+To resolve and validate a Quai CAIP-10 address:
+
+1. **Parse the chain reference** to determine network (9=mainnet, 15000=testnet)
+2. **Extract the address prefix** (first byte) to determine shard routing
+3. **Validate [EIP-55] checksum** using standard Ethereum checksumming
+4. **Route to appropriate RPC endpoint** based on shard location
+
+**Shard Resolution Example**:
+
+```javascript
+// Address: 0x00841c4349dbC3CD16098ef0aF2780BA0F7f5C02
+const firstByte = 0x00;
+const region = (firstByte & 0xF0) >> 4;  // 0
+const zone = firstByte & 0x0F;           // 0
+const isQi = (secondByte & 0x80) > 0;    // false (Quai ledger)
+// Routes to: cyprus-1 (region-0, zone-0)
+```
+
+**RPC Validation**:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "quai_getBalance",
+  "params": ["0x00841c4349dbC3CD16098ef0aF2780BA0F7f5C02", "latest"],
+  "id": 1
+}
+```
+
+**Response**:
+
+```json
+{
+  "jsonrpc": "2.0", 
+  "result": "0x1b1ae4d6e2ef500000",
+  "id": 1
+}
+```
+
+**Shard-to-Port Mapping**:
+
+When running a local Quai node, each shard operates on a specific port.
+The address prefix determines which shard (and thus which port) to use:
+
+| Shard      | HTTP RPC Port | Address Prefix | Status |
+|------------|---------------|----------------|---------|
+| `zone-0-0` (cyprus-1) | 9200 | `0x00`, `0x01` | Active |
+| `zone-0-1` (cyprus-2) | 9201 | `0x10`, `0x11` | Inactive |
+| `zone-1-0` (paxos-1)  | 9220 | `0x20`, `0x21` | Inactive |
+| `zone-1-1` (paxos-2)  | 9221 | `0x30`, `0x31` | Inactive |
+| ... | ... | ... | Future expansion |
+
+**Note**: When using quais.js and/or Pelagus wallet with the mainnet RPC, developers and users are abstracted away from the shard-port mapping.
+If developers run their own node, the shard-port mapping may be a useful reference.
+
+### Backwards Compatibility
+
+Quai addresses maintain full backwards compatibility with Ethereum address formats and tooling:
+
+- **EIP-55 Checksumming**: All Quai addresses use standard Ethereum checksumming
+- **20-byte Format**: Identical to Ethereum address length and structure
+- **Hex Encoding**: Standard 0x-prefixed hex representation
+- **EVM Compatibility**: Addresses work directly with existing Ethereum tooling
+
```

Comment:
does this include UTXO addresses, which are _expressed as_ ethereum addresses? Should EVM developers be careful not to send ethereum "balance" to a UTXO account? it feels like there might be a corner-case to warn people about here, unless this case is caught on the Quai side magically/deterministically!

Reply:
Good catch! Added a note to warn developers not to send to UTXO addresses